sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/HashUtils.scala

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf
+package crypto
+
+import com.daml.scalautil.Statement.discard
+import com.digitalasset.daml.lf.crypto.Hash.underlyingHashLength
+import com.digitalasset.daml.lf.data.Bytes
+
+import java.io.OutputStream
+import java.nio.ByteBuffer
+import java.security.MessageDigest
+
+object HashUtils {
+
+  private val lineSeparator = System.getProperty("line.separator")
+  private[crypto] def formatByteToHexString(byte: Byte): String = String.format("%02X", byte)
+
+  /** Extension of OutputStream with additional methods to help exporting the encoding with context information
+    */
+  abstract class ContextAwareOutputStream extends OutputStream {
+    def write(byte: Byte): Unit
+    def writeContext(context: String): Unit
+  }
+
+  /** Implementation of ContextAwareOutputStream for debugging.
+    * Support contextual info and writes the encoding to a String
+    */
+  class DebugStringOutputStream extends ContextAwareOutputStream {
+    private val sb = new StringBuilder()
+    def result: String = sb.result()
+    override def write(b: Int): Unit =
+      discard(sb.append(Bytes.fromByteArray(Array(b.toByte)).toHexString))
```

Comment:
`Array(b.toByte)` doesn't look right. This will retain only the lowest byte and create a 1-element array. Did you intend to override the `write(b: Byte)` method instead of `write(b: Int)`?